src/profile-logic/process-profile.js

```diff
@@ -1052,56 +1056,195 @@ export function processProfile(
+        stringArray[stringIndex] = stringArray[stringIndex].replace(/:.*/, '');
+      }
+
+      // If PIIToBeRemoved.screenshots is true, this means that we need to
+      // remove all screenshots, if it's an array of thread IDs, that means we
+      // should only delete screenshots from the given threads.
+      if (
+        (PIIToBeRemoved.screenshots === true ||
+          (Array.isArray(PIIToBeRemoved.screenshots) &&
+            PIIToBeRemoved.screenshots.includes(threadIndex))) &&
+        currentMarker &&
+        currentMarker.type &&
+        currentMarker.type === 'CompositorScreenshot'
+      ) {
+        const urlIndex = currentMarker.url;
+        stringArray[urlIndex] = '';
+        markersToDelete.push(i);
+      }
+    }
+  }
+
+  // After iterating (or not iterating at all) the markers, if we have some
+  // markers we want to delete or user wants to delete the full time range,
+  // reconstruct the marker table and samples table without unwanted information.
+  if (markersToDelete.length > 0 || PIIToBeRemoved.fullTimeRange !== null) {
+    const oldMarkers = thread.markers;
+    const newMarkerTable = getEmptyRawMarkerTable();
+    const addCurrentMarker = (index: number) => {
+      if (markersToDelete.includes(index)) {
+        return;
+      }
+      newMarkerTable.name.push(oldMarkers.name[index]);
+      newMarkerTable.time.push(oldMarkers.time[index]);
+      newMarkerTable.data.push(oldMarkers.data[index]);
+      newMarkerTable.length++;
+    };
+
+    // If user wants to remove full time range, filter all the markers and
+    // samples accordingly.
+    if (PIIToBeRemoved.fullTimeRange !== null) {
+      const { start, end } = PIIToBeRemoved.fullTimeRange;
+      const filteredMarkerIndexIter = filterRawMarkerTableIndexesToRange(
+        oldMarkers,
+        start,
+        end
+      );
+
+      for (const index of filteredMarkerIndexIter) {
+        addCurrentMarker(index);
+      }
+
+      // While we are here, we are also filterig the thread samples
+      // to range.
+      if (PIIToBeRemoved.fullTimeRange !== null) {
+        thread = (filterThreadSamplesToRange(thread, start, end): any);
+      }
+    } else {
+      // If user doesn't want to filter out the full time range, remove only
+      // markers that we want to remove.
+      for (let i = 0; i < oldMarkers.length; i++) {
+        addCurrentMarker(i);
+      }
+    }
+
+    thread.markers = newMarkerTable;
+  }
+
+  // This is expensive but needs to be done somehow.
+  // Maybe we can find something better here.
+  if (PIIToBeRemoved.allUrls) {
+    for (let i = 0; i < stringArray.length; i++) {
+      // This regex is complex but so do removing URLs! We had to use much more
+      // complex regex here bacause we have so many edge cases(like different
+      // protocols(moz-extension://) and URLs inside braces without a space in
+      // the middle etc.).
+      stringArray[i] = stringArray[i].replace(
```

Comment:
I'll have more follow-up comments, I'm running out of time for this review, but this regex needs better documentation, and should probably be broken out into a separate function that is battle-tested with some good unit tests.

Reply:
I'm changing this actually. I created this to delete all protocols but it seems like we should keep urls like `chrome://`. It's gonna be more simplified.